ext/cl_khr_command_buffer_mutable_dispatch.asciidoc

```diff
@@ -233,6 +237,7 @@ CL_DEVICE_MUTABLE_DISPATCH_CAPABILITIES_KHR        0x12B0
 
 // Property to cl_ndrange_kernel_command_properties_khr
 CL_MUTABLE_DISPATCH_UPDATABLE_FIELDS_KHR           0x12B1
+CL_MUTABLE_DISPATCH_ASSERTS_KHR                    0x12B7
```

Comment:
This is actually used for two purposes right now:

1. It can be passed as part of the `cl_command_buffer_properties_khr` during command buffer creation to specify properties for the entire command buffer.
2. It can be passed as part of the `cl_ndrange_kernel_command_properties_khr` to specify properties for the specific ND-range kernel recorded into the command buffer.

Would it be better to define two different properties for these two different usages?

Note that the current naming is consistent with (2).  To be consistent with the naming for (1) I think the property should begin with `CL_COMMAND_BUFFER...`, like `CL_COMMAND_BUFFER_FLAGS_KHR`.

Reply:
I agree.

Reply:
> I agree, but is there a way to reuse the new definitions for both usages?

Sure, I don't see why not: The two properties would be different so we would have one property for each purpose, but the value of each property could be the same since they are both defining mutable dispatch assertions.

Reply:
More detail: I propose that we split `CL_MUTABLE_DISPATCH_ASSERTS_KHR` into two properties.  One of these properties can re-use the value `0x12B7` currently assigned to `CL_MUTABLE_DISPATCH_ASSERTS_KHR` - I don't particularly care which one:

```c++
// This is a property that may be specified when the command buffer is created.
// It specifies mutable dispatch assertions for the entire command buffer.

/* cl_command_buffer_properties_khr */
#define CL_COMMAND_BUFFER_MUTABLE_DISPATCH_ASSERTS_KHR   0xXXXX

// This is a property that may be specified for a single ND-range kernel command.
// It specifies mutable dispatch assertions for that ND-range kernel command only.

/* cl_ndrange_kernel_command_properties_khr */
#define CL_MUTABLE_DISPATCH_ASSERTS_KHR                  0xYYYY

// The value for both properties is the same and is a bitfield of mutable dispatch
// assertions.  We will define one bit in the bitfield currently, to assert that no
// additional work-groups will be required.

typedef cl_bitfield         cl_mutable_dispatch_asserts_khr;

/* cl_mutable_dispatch_asserts_khr - bitfield */
#define CL_MUTABLE_DISPATCH_ASSERT_NO_ADDITIONAL_WORK_GROUPS_KHR (1 << 0)
```